### PR TITLE
Fix when enter a URL in button block

### LINKED ISSUE #6841: Button Block does not prefix a URL with `http://`
**Describe the bug**
I had a feeling that this issue issue may have been raised before but I can't find it.

When you enter a url for a button, you can't hit the `Return` key to apply your changes.

On top of that, if you type a url without the `https://` it displays a random result from your site and when you hit `Return`, it actually selects that result instead of using the url you typed

**To Reproduce**
1. Add Button Block
2. Type `google.com` into the URL field
3. Hit `return` key to apply changes

**Expected behavior**
I should be able to simply enter a URL and hit Enter/Return to apply the changes

**Screenshots**
gutenberg_buttonblock

_macOS Sierra 10.12.1 (16B2659)
Firefox Quantum 60.0 (64-bit)
WordPress 4.9.6
Gutenberg 2.9.0_

### PR DESCRIPTION
**Pull request by Rahmon**
## Description

UrlInput component was selecting the first search result in Button Block (and probably in other blocks). 

This happened because onKeyDown event at input just check if some suggestion was selected when Enter key is press.

screenshot from 2018-05-23 11-15-34

Consequently, how suggested posts are buttons inside a form, the onClick event of first button get fired and set the link.

screenshot from 2018-05-23 11-20-34

I also moved prependHTTP function from format-toolbar.js to url-input.js. Otherwise I need to repeat the same function in Button Block.

Closes: #6841 

## How has this been tested?
This has been tested with "npm test" and manually on Chrome

## Types of changes
Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation.

### CODE REVIEW

**Comment by brandonpayton** [comment]
Hi @Rahmon, this seems like a reasonable fix but needs rebased. If you're up for that, I can review after.

**Comment by Rahmon** [comment]
Hi @brandonpayton. I'll update this PR to check if this issue still happening.

**Comment by brandonpayton** [comment]
Some additional info:
On [further discussion](https://github.com/WordPress/gutenberg/issues/6841#issuecomment-422055624), it would be good to get input from design for a full fix to #6841.

**Comment by gziolo** [comment]
@designsimply is that still an issue? I see that all linked references are already closed.

**Comment by designsimply** [comment]
I re-tested and found that the problem raised in #6841 where you cannot hit enter to accept a domain added to the URL input in the link inserter for the button block is still a problem.

#6841 was closed to consolidate with #1838 which was a different issue requesting that validation and feedback be added for the URL input field and which was fixed by #11835 and two additional PRs were split out to address separate concerns about visual formatting for invalid links in #11793 and #11796.

I re-opened #6841 because it has not yet been addressed and added a note to say updating the button block link inserter to use the `URLPopover` as requested in #10128 should solve that problem automatically wince hitting enter to accept a link is already working in `URLPopover` from what I can see in testing.

In summary, I think we can close this PR in favor of updating to `URLPopover` in #10128 and also say thank you to @Rahmon for submitting a fix in the mean time! @Rahmon if you have any additional input or spot something I've missed, please let me know!
